ThreeDSecure/src/main/java/com/braintreepayments/api/ThreeDSecureActivity.java

```diff
@@ -84,10 +84,13 @@ private void finishWithError(String errorMessage) {
     // TODO: NEXT_MAJOR_VERSION remove implementation of CardinalValidateReceiver
     @Override
     public void onValidated(Context context, ValidateResponse validateResponse, String jwt) {
-        handleValidated(validateResponse, jwt);
+        handleValidated(cardinalClient, validateResponse, jwt);
     }
 
-    private void handleValidated(ValidateResponse validateResponse, String jwt) {
+    @VisibleForTesting
+    void handleValidated(CardinalClient cardinalClient, ValidateResponse validateResponse, String jwt) {
+        cardinalClient.cleanup();
```

Comment:
Should we check if `cardinalClient` is null here? Not sure if it could be but better to safeguard

Reply:
We should be good it's marked final [here](https://github.com/braintree/braintree_android/blob/6f8a39c8642b01e385aeed591cefb0b048326108/ThreeDSecure/src/main/java/com/braintreepayments/api/ThreeDSecureActivity.java#L29).